src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java

```diff
@@ -37,35 +42,84 @@ public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFact
     this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
   }
 
-  public void setSdkDownloader(CloudSdkDownloader downloader) {
-    this.downloader = downloader;
+  public void setManagedCloudSdkFactory(ManagedCloudSdkFactory managedCloudSdkFactory) {
+    this.managedCloudSdkFactory = managedCloudSdkFactory;
   }
 
   /** Task entrypoint : Download/update/verify Cloud SDK installation. */
   @TaskAction
-  public void downloadCloudSdkAction() {
+  public void downloadCloudSdkAction() throws Exception {
```

Comment:
I think we tend to avoid using the superclass `Exception` and just include all the individual exceptions.

Reply:
There are a lot of exceptions that can be thrown (IntelliJ's autocomplete shortened the list to just "Exception"), should I still add individual exceptions even when the list is really long?

Reply:
Hmm, not sure then. @loosebazooka @elharo ?

Reply:
This code is launched by the gradle system, all exception get wrapped as GradleExceptions or TaskException or something I think and effectively end the build process? We'll never really be handling what's thrown here, so I believe there is loss of information. If on the other hand we did want to handle everything then we should have the appropriate try/catch blocks in the code here.

Reply:
Yes, I definitely prefer explicit exception lists where possible. It's OK to wrap lower level exceptions in a supertype if the supertype is itself thrown. For instance, you don't need to declare FileNotFoundException if a raw IOException can also be thrown. However you shouldn't declare that IOException can be thrown when it isn't. java.lang.Exception should generally not be declared excpet in very special circumstances such as Callable or Runnable where anything can happen, and probably not then.